pkg/vsphere/disk/vmdk.go

```diff
@@ -58,7 +64,34 @@ func LockedVMDKFilter(vm *mo.VirtualMachine) bool {
 	return vm.Runtime.PowerState == types.VirtualMachinePowerStatePoweredOn
 }
 
-func IsLockedError(err error) bool {
-	// TODO: add check here for actual error
-	return true
+func IsLockedError(op trace.Operation, err error) bool {
+	switch err := err.(type) {
+	case task.Error:
+		fault := err.Fault().GetMethodFault()
+
+		// indicates it's specific to a disk
+		diskBackend := false
+		// indicates it's specific to locking
+		fileLock := false
+
+		for i := range fault.FaultMessage {
+			message := &fault.FaultMessage[i]
+			switch message.Key {
+			case DiskBackendKey:
+				op.Debugf("diskbackend is true")
```

Comment:
If we're logging the original error then we don't need these debug messages - I'm wishing again we had more debug levels as this is fine grained trace.

Can you consolidate the debugging messages so we get a single log on exit that tells us the details.